pkg/build/api/v1/types.go

```diff
@@ -402,6 +408,10 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// ExternalBuildStrategy defines input parameters specific to an External build.
+type ExternalBuildStrategy struct {
```

Comment:
per my other comment, the buildconfig should not be updated by an instance of the job, there should be a build object created and that should contain the url of the specific job.

as for annotations vs fields, yes i think we want fields..something like:

```
type ExternalBuildStrategy struct {
  Type string // the type of external strategy, eg "jenkins"
  Jenkins *JenkinsStrategy
  Travis *TravisStrategy
}

type JenkinsStrategy struct {
  JobURL string
  Credentials *LocalObjectReference // reference to a Secret
  // etc etc
}
```